sys/include/xtimer/tick_conversion.h

```diff
@@ -12,22 +13,85 @@
  * @{
  * @file
  * @brief   xtimer tick <-> seconds conversions for different values of XTIMER_HZ
+ *
  * @author  Joakim Nohlgård <joakim.nohlgard@eistec.se>
+ * @author  Josua Arndt <jarndt@ias.rwth-aachen.de>
  */
-
 #ifndef XTIMER_TICK_CONVERSION_H
 #define XTIMER_TICK_CONVERSION_H
 
 #ifndef XTIMER_H
 #error "Do not include this file directly! Use xtimer.h instead"
 #endif
 
+#include <stdio.h>
+#include <string.h>
+#include "fmt.h"
 #include "div.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+/*
+ * As tick_conversion.h is included in multiple files ENABLE_DEBUG can not be
+ * used as it gets redefined "[-Werror]"
+ * Activate the debug print and enable module fmt in the project makefile as follow
+ *
+ * USEMODULE += fmt
+ * CFLAGS += -DENABLE_DEBUG_PRINT_TICK_CONVERSION=1
+ * */
+#ifndef ENABLE_DEBUG_PRINT_TICK_CONVERSION
+#define ENABLE_DEBUG_PRINT_TICK_CONVERSION   (0)
+#endif
+
+#if ENABLE_DEBUG_PRINT_TICK_CONVERSION
+#ifndef MODULE_FMT
+#error Module fmt is necessary to ENABLE_DEBUG_TICKS
+/* Add module fmt to the makefile.
+ * USEMODULE += fmt
+ */
+#endif
+#endif
+
+inline void debug_print_ticks32(uint32_t clock_hz, char *sign, uint32_t from,
+                         char *base_from, uint32_t to,char *base_to) {
+    if(ENABLE_DEBUG_PRINT_TICK_CONVERSION){
+        print_str("32bit ");
+        print_u32_dec(clock_hz);
+        print_str(" ");
+        print_str(sign);
+        print_str(" 1MHz : ");
+        print_u32_dec(from);
+        print_str(" ");
+        print_str(base_from);
+        print_str(" = ");
+        print_u32_dec(to);
+        print_str(" ");
+        print_str(base_to);
+        print_str("\n");
+    }
+}
+
+inline void debug_print_ticks64(uint32_t clock_hz, char *sign, uint64_t from,
+                         char *base_from, uint64_t to,char *base_to) {
```

Comment:
dito, see above

Reply:
though I see below, that there are some diversion from the coding style here anyway, i.e. parentheses not on a new line.

Reply:
i added a seperate uncrustify commit, which also adresses this.